mono/utils/mono-error.h

```diff
@@ -49,11 +49,13 @@ enum {
 };
 
 /*Keep in sync with MonoErrorInternal*/
-typedef struct _MonoError {
-	unsigned short error_code;
-    unsigned short hidden_0; /*DON'T TOUCH */
-
-	void *hidden_1 [12]; /*DON'T TOUCH */
+typedef union _MonoError {
+	uint32_t init; // merge two uint16 so they can be written with one instruction
```

Comment:
What's the point of this `init` field if we access it at only one place? It seems to complexify the struct just to ease the implementation of `error_init(_internal)`.

Reply:
It an optimization, guarantee one instruction to do the two stores. I found x86/amd64/arm64 made one store anyway, but arm32 did not. The comment explains it, I thought. This isn't particularly complicated also I think, at least with the comment.